apps/meteor/app/livechat/server/business-hour/BusinessHourManager.ts

```diff
@@ -232,4 +238,42 @@ export class BusinessHourManager {
 	private clearCronJobsCache(): void {
 		this.cronJobsCache = [];
 	}
+
+	hasDaylightSavingTimeChanged(timezone: IBusinessHourTimezone): boolean {
+		const now = moment().utc().tz(timezone.name);
+		const currentUTC = now.format('Z');
+		const existingTimezoneUTC = moment(timezone.utc, 'Z').utc().tz(timezone.name);
+		const DSTHasChanged = !moment(currentUTC, 'Z').utc().tz(timezone.name).isSame(existingTimezoneUTC);
+
+		return currentUTC !== timezone.utc && ((now.isDST() && DSTHasChanged) || (!now.isDST() && DSTHasChanged));
+	}
+
+	async registerDaylightSavingTimeCronJob(): Promise<void> {
+		await this.cronJobs.add(DAYLIGHT_CRON_JOB_NAME, EVERY_MIDNIGHT_CRON_EXPRESSION, this.startDaylightSavingTimeVerifier.bind(this));
+	}
+
+	async startDaylightSavingTimeVerifier(): Promise<void> {
+		const activeBusinessHours = await LivechatBusinessHours.findActiveBusinessHours();
+		const timezonesNeedingAdjustment = activeBusinessHours.filter(
+			({ timezone }) => timezone && this.hasDaylightSavingTimeChanged(timezone),
+		);
+		if (timezonesNeedingAdjustment.length === 0) {
+			return;
+		}
+		await Promise.all(
+			timezonesNeedingAdjustment.map((businessHour) => {
+				const businessHourType = this.getBusinessHourType(businessHour.type) as IBusinessHourType;
+
+				return businessHourType.saveBusinessHour({
+					...businessHour,
+					timezoneName: businessHour.timezone.name,
+					workHours: businessHour.workHours.map((hour) => ({ ...hour, start: hour.start.time, finish: hour.finish.time })) as Record<
+						string,
+						any
+					>[],
+				} as ILivechatBusinessHour & { timezoneName: string });
+			}),
+		);
```

Comment:
would it make sense here to have as many changes done without failing (all vs allSettled)?

Reply:
For sure, thanks for noticing that, I'll change